### PR TITLE
MP-CNN Bugfixes and Improvements

### PR DESCRIPTION
**Pull request by tuzhucheng**
Some bugfixes and improvements for MP-CNN. Results on SICK exceeds paper performance and results on MSRVID gets very close.

* Fix bug in pooling - previously pooling kernel size for holistic conv output was specified incorrectly - and code would crash if number using a non-default number of holistic filters that's not 300
* Results from infinite `ws` are now properly pooled
* Ability to configure optimizer (Adam or SGD), patience, learning rate reduce factor after plateau

SICK dataset

| Implementation and config        | Pearson's r   | Spearman's p  |
| -------------------------------- |:-------------:|:-------------:|
| Paper                            | 0.8686        |   0.8047      |
| PyTorch using README config       | 0.8763        |   0.8215      |

MSRVID dataset

| Implementation and config        | Pearson's r   |
| -------------------------------- |:-------------:|
| Paper                            | 0.9090        |
| PyTorch using README config       | 0.9050        |

@Impavidity please review?

### CODE REVIEW

**Comment by Impavidity** [comment]
LGTM.
